pootle/i18n/formatter.py

```diff
@@ -15,12 +15,14 @@
 from django.utils.translation import get_language, to_locale
 
 
-def get_locale_formats():
-    for language in [get_language(), settings.LANGUAGE_CODE, 'en-us']:
+def get_locale_formats(locale=None):
+    languages = [locale] if locale else []
+    languages += [get_language(), settings.LANGUAGE_CODE, 'en-us', 'en']
+    for language in languages:
         try:
             locale = babel_core.Locale.parse(to_locale(language))
             break
-        except UnknownLocaleError:
+        except (UnknownLocaleError, ValueError):
```

Comment:
It seems `babel_support.Format(babel_core.Locale.parse(to_locale('en-us'))).timedelta(timedelta, format='long')` is not the same as `format_timedelta(timedelta, locale='en-us')`. The latter actually yields a `ValueError` and thus the default to `en`. Changed PR to remove all of that.